x/validator/handler_test.go

```diff
@@ -177,3 +186,360 @@ func queryValidator(setup TestSetup, owner string) (*types.Validator, error) {
+
+	ba3 := authtypes.NewBaseAccount(testconstants.Address3, testconstants.PubKey3, 0, 0)
+	account3 := dclauthtypes.NewAccount(ba3,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID3)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account3)
+
+	valAddr := sdk.ValAddress(testconstants.ValidatorAddress1)
+
+	// propose and approve new disablevalidators
+	msgProposeDisableValidator1 := NewMsgProposeDisableValidator(account1.GetAddress(), valAddr)
+	_, err := setup.Handler(setup.Ctx, msgProposeDisableValidator1)
+	require.NoError(t, err)
+
+	proposedDisableValidator, isFound := setup.ValidatorKeeper.GetProposedDisableValidator(setup.Ctx, valAddr.String())
+	require.True(t, isFound)
+	require.Equal(t, msgProposeDisableValidator1.Address, proposedDisableValidator.Address)
+	require.Equal(t, msgProposeDisableValidator1.Creator, proposedDisableValidator.Creator)
+	require.Equal(t, msgProposeDisableValidator1.Creator, proposedDisableValidator.Approvals[0].Address)
+
+	_, isFound = setup.ValidatorKeeper.GetDisabledValidator(setup.Ctx, valAddr.String())
+	require.False(t, isFound)
+
+	msgApproveDisableValidator := NewMsgApproveDisableValidator(account2.GetAddress(), valAddr)
+	_, err = setup.Handler(setup.Ctx, msgApproveDisableValidator)
+	require.NoError(t, err)
+
+	_, isFound = setup.ValidatorKeeper.GetProposedDisableValidator(setup.Ctx, valAddr.String())
+	require.False(t, isFound)
+
+	disabledValidator, isFound := setup.ValidatorKeeper.GetDisabledValidator(setup.Ctx, valAddr.String())
+	require.True(t, isFound)
+	require.Equal(t, msgProposeDisableValidator1.Address, disabledValidator.Address)
+	require.Equal(t, msgProposeDisableValidator1.Creator, disabledValidator.Creator)
+	require.Equal(t, msgProposeDisableValidator1.Creator, disabledValidator.Approvals[0].Address)
+	require.Equal(t, msgProposeDisableValidator1.Info, disabledValidator.Approvals[0].Info)
+	require.Equal(t, msgProposeDisableValidator1.Time, disabledValidator.Approvals[0].Time)
+	require.False(t, disabledValidator.DisabledByNodeAdmin)
+
+	validator, isFound := setup.ValidatorKeeper.GetValidator(setup.Ctx, valAddr)
+	require.True(t, isFound)
+	require.True(t, validator.Jailed)
+}
+
+func TestHandler_DisabledValidatorOnPropose(t *testing.T) {
+	setup := Setup(t)
+
+	// create Trustees
+	ba1 := authtypes.NewBaseAccount(testconstants.Address1, testconstants.PubKey1, 0, 0)
+	account1 := dclauthtypes.NewAccount(ba1,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID1)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account1)
+
+	ba2 := authtypes.NewBaseAccount(testconstants.Address2, testconstants.PubKey2, 0, 0)
+	account2 := dclauthtypes.NewAccount(ba2,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID2)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account2)
+
+	ba3 := authtypes.NewBaseAccount(sdk.AccAddress(testconstants.ValidatorAddress1), testconstants.PubKey3, 0, 0)
```

Comment:
The conversion of the address is incorrect (here and below).

`ValidatorAddress1` is a Bech32 `cosmosvaloper` address. To convert it to `AccAddress` the following steps should be done:
- function `ValAddressFromBech32` should be used to convert the original value to `ValAddress`,
- the conversion `AccAddress(...)` should be used to convert the result of the previous step to `AccAddress`.